### PR TITLE
Sidebar sections integration

### PR DESCRIPTION
**Pull request by juliusknorr**
- Implements projects section in the sharing sidebar
- Add ShareTabSections service to register sections in the share tab
  - This will be used by groupfolders for the ACL management UI https://github.com/nextcloud/groupfolders/pull/507
  - Another use case would be the list of current guest editors added to the sharing sidebar, which is something @jancborchardt wished for quite some time :wink: 

Requires a new build of nextcloud-vue-collections

### CODE REVIEW

File: apps/files_sharing/src/views/SharingTab.vue [hunk 1]
```diff
@@ -33,6 +33,12 @@
 			<SharingEntry v-if="isSharedWithMe" v-bind="sharedWithMe"></SharingEntry>
 			<SharingInput :shares="shares" :file-info="fileInfo" :reshare="reshare" />
 		</template>
+
+		<div class="sharingTab__additionalContent" v-for="(section, $index) in sections" :ref="'section-' + $index">
+			<div :is="section($refs['section-'+$index], fileInfo)" :file-info="fileInfo"></div>
+		</div>
+
+		<collection-list v-if="fileInfo" type="file" :id="`${fileInfo.id}`" :name="fileInfo.name"></collection-list>
```
Comment (skjnldsv) [inline apps/files_sharing/src/views/SharingTab.vue]: Do you need it as string?

Comment (juliushaertl) [reply apps/files_sharing/src/views/SharingTab.vue]: Yes, the id of a project resource is of type string, so we need to cast.

**Comment by juliushaertl** [comment]
@skjnldsv I leave bumping the nextcloud-vue-collections to 0.5.4 up to you if thats fine, so we have a bit less rebasing work here.

**Comment by MorrisJobke** [comment]
@juliushaertl What is the status here? We are close to the beta 1. Should this go into 17 or 18?

**Comment by juliushaertl** [comment]
18 since this is part of https://github.com/nextcloud/server/pull/15719

**Pull request closed without merging**